lib/read.js

```diff
@@ -136,13 +139,17 @@ class ReadInflux extends AdapterRead {
         return this.queryOptions.raw === undefined;
     }
 
-    read(from, to, limit, state) {
+    read(from, to, batchSize, state) {
         var self = this;
 
         var parsedUrl = _.clone(this.url);
         var reqUrl = null;
 
-        var queryOptions = _.extend({ to, from }, this.queryOptions);
+        // If scheduler is throttling our reads via batchSize param, take it,
+        // even if there are more points left to read.
+        var limit = Math.min(batchSize, this.queryOptions.limit - this.pointsRead);
```

Comment:
I think most other adapters use the `fetchSize` option. Consider renaming to the `limit` option here to `fetchSize`.